spec/compiler/macro/macro_methods_spec.cr

```diff
@@ -1866,6 +1866,134 @@ module Crystal
         end
       end
 
+      describe "#constant" do
+        it "global path" do
+          assert_type(%(
+            ID = 10
+
+            class A
+              class B
+              end
+            end
+
+            {{ A::B.constant("::ID") == 10 ? 1 : 'f' }}
+          )) { int32 }
+        end
+
+        it "global path with extra ::" do
+          expect_raises(Crystal::TypeException, %(Invalid constant name: "::::ID")) do
+            assert_macro %(ID = 10; {{@type.has_constant? "::::ID" }}), %(nil)
```

Comment:
There's no point in adding additional types, since `has_constant?` will raise regardless:

```suggestion
            assert_macro %({{ @type.has_constant? "::::ID" }}), "nil"
```

Reply:
There is an `assert_macro_error` macro for this